pinot-query-runtime/src/test/resources/queries/FromExpressions.json

```diff
@@ -0,0 +1,221 @@
+          [1, "xxx"],
+          [3, "yyy"],
+          [5, "zzz"]
+        ]
+      }
+    },
+    "queries": [
+      {
+        "psql": "7.2.1.1",
+        "description": "Basic From table",
+        "sql": "SELECT * FROM {tbl1}"
+      },
+      {
+        "psql": "7.2.1.1",
+        "description": "Basic From table with Alias",
+        "sql": "SELECT a.num, a.name FROM {tbl1} AS a"
+      },
+      {
+        "psql": "7.2.1.1",
+        "description": "INNER JOIN",
+        "sql": "SELECT * FROM {tbl1} INNER JOIN {tbl2} ON {tbl1}.num = {tbl2}.num"
+      },
+      {
+        "psql": "7.2.1.1",
+        "description": "LEFT OUTER JOIN",
+        "sql": "SELECT * FROM {tbl1} LEFT OUTER JOIN {tbl2} ON {tbl1}.num = {tbl2}.num"
+      },
+      {
+        "psql": "7.2.1.1",
+        "description": "LEFT OUTER JOIN with non-join related clause",
+        "sql": "SELECT * FROM {tbl1} LEFT OUTER JOIN {tbl2} ON {tbl1}.num = {tbl2}.num AND {tbl2}.value = 'xxx'"
+      },
+      {
+        "psql": "7.2.1.1",
+        "description": "CROSS JOIN",
+        "sql": "SELECT * FROM {tbl1} CROSS JOIN {tbl2}"
+      },
+      {
+        "psql": "7.2.1.1",
+        "ignored": true,
+        "description": "RIGHT OUTER JOIN",
+        "sql": "SELECT * FROM {tbl1} RIGHT OUTER JOIN {tbl2} ON {tbl1}.col1 = {tbl2}.col1"
+      },
+      {
+        "psql": "7.2.1.1",
+        "ignored": true,
+        "description": "FULL OUTER JOIN",
+        "sql": "SELECT * FROM {tbl1} FULL OUTER JOIN {tbl2} ON {tbl1}.col1 = {tbl2}.col1"
+      },
+      {
+        "psql": "7.2.1.1",
+        "description": "NATURAL JOIN",
+        "sql": "SELECT * FROM {tbl1} NATURAL JOIN {tbl2}"
+      },
+      {
+        "psql": "7.2.1.2",
+        "description": "JOIN with table alias",
+        "sql": "SELECT * FROM {tbl1} a LEFT JOIN {tbl2} b ON a.num = b.num"
+      },
+      {
+        "psql": "7.2.1.2",
+        "description": "self JOIN with table alias",
+        "sql": "SELECT * FROM {tbl1} AS a INNER JOIN {tbl1} AS b ON a.num = b.num"
+      },
+      {
+        "psql": "7.2.1.2",
+        "description": "table alias on one side only using star",
+        "sql": "SELECT a.* FROM {tbl1} AS a JOIN {tbl2} AS b ON a.num = b.num"
+      },
+      {
+        "psql": "7.2.1.3",
+        "description": "sub-query to semi-join syntax with star results, using IN clause",
+        "sql": "SELECT * FROM {tbl1} WHERE num IN (SELECT num FROM {tbl2})"
+      },
+      {
+        "psql": "7.2.1.3",
+        "description": "sub-query to semi-join syntax with star results, using single value predicate clause",
+        "sql": "SELECT * FROM {tbl1} WHERE num < (SELECT SUM(num) FROM {tbl2})"
+      },
+      {
+        "psql": "7.2.1.3",
+        "description": "sub-query used as one side of the join condition",
+        "sql": "SELECT * FROM (SELECT num AS id, name FROM {tbl1} WHERE name != 'c') AS a JOIN {tbl2} AS b ON a.id = b.num"
+      }
+    ]
+  },
+  "extended_join_features": {
```

Comment:
Can we have a test where left or right table is empty?
Can we add join with using condition and where exists?

Reply:
- added several where clauses that matches nothing on left or right table already. but yes I can add several explicit joins where left or right table is entirely empty
- where exists is not supported.

Reply:
Hmm...last time I tested where exists, it was supported. Let me double check

Reply:
where exists is also added in https://github.com/apache/pinot/pull/9817 and ignored.

Reply:
added empty table explicit tests